integrations/slack/client.go

```diff
@@ -34,230 +37,257 @@ var desc = kittehs.Must1(sdktypes.StrictIntegrationFromProto(&sdktypes.Integrati
 	},
 }))
 
-func New(vars sdkservices.Vars) sdkservices.Integration {
-	authAPI := auth.API{Vars: vars}
-	bookmarksAPI := bookmarks.API{Vars: vars}
-	botsAPI := bots.API{Vars: vars}
-	chatAPI := chat.API{Vars: vars}
-	conversationsAPI := conversations.API{Vars: vars}
-	reactionsAPI := reactions.API{Vars: vars}
-	usersAPI := users.API{Vars: vars}
-
+func New(vs sdkservices.Vars) sdkservices.Integration {
 	return sdkintegrations.NewIntegration(
 		desc,
-		sdkmodule.New(
-			// Auth.
-			sdkmodule.ExportFunction(
-				"auth_test",
-				authAPI.Test,
-				sdkmodule.WithFuncDoc("https://api.slack.com/methods/auth.test"),
-			),
+		sdkmodule.New(funcs(vs)...),
+		connStatus(vs),
+		sdkintegrations.WithConnectionConfigFromVars(vs),
+	)
+}
 
-			// Bookmarks.
-			sdkmodule.ExportFunction(
-				"bookmarks_add",
-				bookmarksAPI.Add,
-				sdkmodule.WithFuncDoc("https://api.slack.com/methods/bookmarks.add"),
-				sdkmodule.WithArgs("channel_id", "title" /* , "type" */, "link?", "emoji?", "entity_id?", "parent_id?"),
-			),
-			sdkmodule.ExportFunction(
-				"bookmarks_edit",
-				bookmarksAPI.Edit,
-				sdkmodule.WithFuncDoc("https://api.slack.com/methods/bookmarks.edit"),
-				sdkmodule.WithArgs("bookmark_id", "channel_id", "emoji?", "link?", "title?"),
-			),
-			sdkmodule.ExportFunction(
-				"bookmarks_list",
-				bookmarksAPI.List,
-				sdkmodule.WithFuncDoc("https://api.slack.com/methods/bookmarks.list"),
-				sdkmodule.WithArgs("channel_id"),
-			),
-			sdkmodule.ExportFunction(
-				"bookmarks_remove",
-				bookmarksAPI.Remove,
-				sdkmodule.WithFuncDoc("https://api.slack.com/methods/bookmarks.remove"),
-				sdkmodule.WithArgs("bookmark_id", "channel_id", "quip_section_id?"),
-			),
+// connStatus is an optional connection status check provided by the
+// integration to AutoKitteh. The possible results are "init required"
+// (the connection is not usable yet) and "using X" (where "X" is the
+// authentication method: an OAuth v2 app, or a Socket Mode app).
+func connStatus(vs sdkservices.Vars) sdkintegrations.OptFn {
+	return sdkintegrations.WithConnectionStatus(func(ctx context.Context, cid sdktypes.ConnectionID) (sdktypes.Status, error) {
+		if !cid.IsValid() {
+			return sdktypes.NewStatus(sdktypes.StatusCodeWarning, "init required"), nil
```

Comment:
We need a formal definition of the state machine of connection. "not valid" can mean also bad credentials.

Reply:
I agree with the need for a formal definition - I'll follow-up with Haim, because this has a strong UX & doc'n aspect more than a technical aspect.

Bad creds: creds are validated for each connection before saving them, so this is not a pressing concern, but you're right that they can still be invalidated later.

When we'll add statefulness to post-creation connection tests and, we will also reuse the existing pre-creation checks code.